pkg/config/config.go

```diff
@@ -73,6 +80,28 @@ func LoadFile(filename string, logger log.Logger) (*Config, []byte, error) {
 	return cfg, content, nil
 }
 
+// expand env variables $(var) from the config file
+// taken from https://github.dev/thanos-io/thanos/blob/296c4ab4baf2c8dd6abdf2649b0660ac77505e63/pkg/reloader/reloader.go#L445-L462 by https://github.com/fabxc
+func substituteEnvVars(b []byte, logger log.Logger) (r []byte, err error) {
+	var envRe = regexp.MustCompile(`\$\(([a-zA-Z_0-9]+)\)`)
+	r = envRe.ReplaceAllFunc(b, func(n []byte) []byte {
+		if err != nil {
+			return nil
+		}
+
+		n = n[2 : len(n)-1]
+
+		v, ok := os.LookupEnv(string(n))
+		if !ok {
+			level.Error(logger).Log("msg", "env variable from config not found", "variable", string(n))
+			err = fmt.Errorf("Missing env variable: %q", n)
+			return nil
```

Comment:
Let's handle error once, here we return it with `err =` and  logging. but it will be logged when `substituteEnvVars` will handle ther error.

Reply:
removed redundant error handling